Wails build creating invalid folders based on embed file selections. Then failing.

### Description

I have a strange problem where the "wails build" process is creating folders with illegal names based on files I am embedding. The embed commands are not part of the wails app but are in another package in my project.

```
handlers/render.go contains:
//go:embed assets/*
var assetsFs embed.FS
//go:embed *.gohtml
//go:embed layouts/*.gohtml
var templateFs embed.FS
```

Wails build creates folders with those name `'assets/*', '*.gohtml'`, etc. with a file gitkeep in them. It's either wails or some component used by wails.

Then, it fails to compile due to those existing. It is definitely wails build that is creating them. I delete them and they show up again.

```
$ find .|grep '*'
./cmd/prog/package/handlers/*.gohtml
./cmd/prog/package/handlers/*.gohtml/gitkeep
./cmd/prog/package/handlers/layouts/*.gohtml
./cmd/prog/package/handlers/layouts/*.gohtml/gitkeep
./cmd/prog/package/handlers/assets/*
./cmd/prog/package/handlers/assets/*/gitkeep
```

### To Reproduce

Just running wails build in my project root. It worked fine for months and just started.

```
$ wails build
Wails CLI v2.8.1
# Build Options
Platform(s)        | linux/amd64                                               
Compiler           | /usr/local/go/bin/go                                      
Skip Bindings      | false                                                     
Build Mode         | production                                                
Devtools           | false                                                     
Frontend Directory | /home/pcarroll/Development/Golang/netrinos-client/frontend
Obfuscated         | false                                                     
Skip Frontend      | false                                                     
Compress           | false                                                     
Package            | true                                                      
Clean Bin Dir      | false                                                     
LDFlags            |                                                           
Tags               | []                                                        
Race Detector      | false                                                     

# Building target: linux/amd64
  • Generating bindings:   ERROR   
          cmd/prog/package/handlers/render.go:27:12: pattern *.gohtml: cannot embed directory *.gohtml: invalid name *.gohtml
          exit status 1
  ERROR   
          cmd/prog/edgenode/handlers/render.go:27:12: pattern *.gohtml: cannot embed directory *.gohtml: invalid name *.gohtml          
          exit status 1
```
### Expected behaviour

It just compiles on Windows and used to on Linux

### Screenshots

_No response_

### Attempted Fixes

I delete the folders, delete the build folder. It all comes back.

### System Details

```shell
On Linux 22.04 amd64 with Wails 2.8.1 and Go 1.22.2 it fails. Same on Windows 11 works fine as before.


$ wails doctor

                                
          Wails Doctor          
                                

                                                                                                                           
# Wails
Version         | v2.8.1
Package Manager | apt   

# System
┌───────────────────────────────────────────────────────────────────────────────────────┐
| OS           | Linux Mint                                                             |
| Version      | 21.3                                                                   |
| ID           | linuxmint                                                              |
| Go Version   | go1.22.2                                                               |
| Platform     | linux                                                                  |
| Architecture | amd64                                                                  |
| CPU          | Intel(R) Core(TM) i7-8700T CPU @ 2.40GHz                               |
| GPU          | CoffeeLake-S GT2 [UHD Graphics 630] (Intel Corporation) - Driver: i915 |
| Memory       | 66GB                                                                   |
└───────────────────────────────────────────────────────────────────────────────────────┘

# Dependencies
┌──────────────────────────────────────────────────────────────────────────┐
| Dependency | Package Name          | Status    | Version                 |
| *docker    | docker.io             | Available | 24.0.5-0ubuntu1~22.04.1 |
| gcc        | build-essential       | Installed | 12.9ubuntu3             |
| libgtk-3   | libgtk-3-dev          | Installed | 3.24.33-1ubuntu2        |
| libwebkit  | libwebkit2gtk-4.0-dev | Installed | 2.44.0-0ubuntu0.22.04.1 |
| npm        | npm                   | Installed | 10.2.4                  |
| *nsis      | nsis                  | Installed | v3.08-2                 |
| pkg-config | pkg-config            | Installed | 0.29.2-1ubuntu3         |
└──────────────────────── * - Optional Dependency ─────────────────────────┘

# Diagnosis
Optional package(s) installation details: 
  - docker: sudo apt install docker.io

 SUCCESS  Your system is ready for Wails development!
```


### Additional context

Please give me some hints on how to debug this...

--- a/v2/internal/staticanalysis/staticanalysis.go
+++ b/v2/internal/staticanalysis/staticanalysis.go
@@ -52,24 +52,31 @@ func GetEmbedDetailsForFile(file *ast.File, baseDir string) []*EmbedDetails {
 		for _, c := range comment.List {
 			if strings.HasPrefix(c.Text, "//go:embed") {
 				sl := strings.Split(c.Text, " ")
-				path := ""
-				all := false
 				if len(sl) == 1 {
 					continue
 				}
-				embedPath := strings.TrimSpace(sl[1])
-				switch true {
-				case strings.HasPrefix(embedPath, "all:"):
-					path = strings.TrimPrefix(embedPath, "all:")
-					all = true
-				default:
-					path = embedPath
+				// support for multiple paths in one comment
+				for _, arg := range sl[1:] {
+					embedPath := strings.TrimSpace(arg)
+					// ignores all pattern matching characters except escape sequence
+					if strings.Contains(embedPath, "*") || strings.Contains(embedPath, "?") || strings.Contains(embedPath, "[") {
+						continue
+					}
+					if strings.HasPrefix(embedPath, "all:") {
+						result = append(result, &EmbedDetails{
+							EmbedPath: strings.TrimPrefix(embedPath, "all:"),
+							All:       true,
+							BaseDir:   baseDir,
+						})
+					} else {
+						result = append(result, &EmbedDetails{
+							EmbedPath: embedPath,
+							All:       false,
+							BaseDir:   baseDir,
+						})
+					}
+
 				}
-				result = append(result, &EmbedDetails{
-					EmbedPath: path,
-					All:       all,
-					BaseDir:   baseDir,
-				})
 			}
 		}
 	}

--- a/v2/internal/staticanalysis/staticanalysis_test.go
+++ b/v2/internal/staticanalysis/staticanalysis_test.go
@@ -1,8 +1,9 @@
 package staticanalysis
 
 import (
-	"github.com/stretchr/testify/require"
 	"testing"
+
+	"github.com/stretchr/testify/require"
 )
 
 func TestGetEmbedDetails(t *testing.T) {
@@ -24,6 +25,10 @@ func TestGetEmbedDetails(t *testing.T) {
 				{
 					EmbedPath: "frontend/dist",
 					All:       true,
+				},
+				{
+					EmbedPath: "frontend/static",
+					All:       false,
 				},
 			},
 			wantErr: false,

--- a/v2/internal/staticanalysis/test/standard/main.go
+++ b/v2/internal/staticanalysis/test/standard/main.go
@@ -8,8 +8,11 @@ import (
 	"github.com/wailsapp/wails/v2/pkg/options/assetserver"
 )
 
-//go:embed all:frontend/dist
+//go:embed all:frontend/dist frontend/static
 var assets embed.FS
+
+//go:embed frontend/src/*.json
+var srcjson embed.FS
 
 func main() {
 	// Create an instance of the app structure

--- a/v2/pkg/commands/build/build.go
+++ b/v2/pkg/commands/build/build.go
@@ -169,16 +169,19 @@ func CreateEmbedDirectories(cwd string, buildOptions *Options) error {
 
 	for _, embedDetail := range embedDetails {
 		fullPath := embedDetail.GetFullPath()
-		if _, err := os.Stat(fullPath); os.IsNotExist(err) {
-			err := os.MkdirAll(fullPath, 0o755)
-			if err != nil {
-				return err
+		// assumes path is directory only if it has no extension
+		if filepath.Ext(fullPath) == "" {
+			if _, err := os.Stat(fullPath); os.IsNotExist(err) {
+				err := os.MkdirAll(fullPath, 0755)
+				if err != nil {
+					return err
+				}
+				f, err := os.Create(filepath.Join(fullPath, "gitkeep"))
+				if err != nil {
+					return err
+				}
+				_ = f.Close()
 			}
-			f, err := os.Create(filepath.Join(fullPath, "gitkeep"))
-			if err != nil {
-				return err
-			}
-			_ = f.Close()
 		}
 	}
 

--- a/v2/pkg/commands/build/build.go
+++ b/v2/pkg/commands/build/build.go
@@ -172,7 +172,7 @@ func CreateEmbedDirectories(cwd string, buildOptions *Options) error {
 		// assumes path is directory only if it has no extension
 		if filepath.Ext(fullPath) == "" {
 			if _, err := os.Stat(fullPath); os.IsNotExist(err) {
-				err := os.MkdirAll(fullPath, 0755)
+				err := os.MkdirAll(fullPath, 0o755)
 				if err != nil {
 					return err
 				}

--- a/website/src/pages/changelog.mdx
+++ b/website/src/pages/changelog.mdx
@@ -25,7 +25,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ### Fixed
 
-- Fixed an issue where embed comments with patterned paths would create illegal directories. Changed by [@twonull](https://github.com/twonull) in [PR](https://github.com/wailsapp/wails/pull/3445)
+- Fixed an issue where embed directives with patterned paths would create illegal directories. Changed by [@twonull](https://github.com/twonull) in [PR](https://github.com/wailsapp/wails/pull/3445)
 - Fixed an issue where certain calls to createFrom in TypeScript would fail. Changed by [@twonull](https://github.com/twonull) in [PR](https://github.com/wailsapp/wails/pull/3435)
 - Fixed some typos in comments. Changed by [@reallylowest](https://github.com/reallylowest) in [PR](https://github.com/wailsapp/wails/pull/3357)
 - Fixed an issue where the destination file was not properly closed after copying. Changed by [@testwill](https://github.com/testwill) in [PR](https://github.com/wailsapp/wails/pull/3384)

--- a/website/src/pages/changelog.mdx
+++ b/website/src/pages/changelog.mdx
@@ -25,6 +25,7 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 
 ### Fixed
 
+- Fixed an issue where embed comments with patterned paths would create illegal directories. Changed by [@twonull](https://github.com/twonull) in [PR](https://github.com/wailsapp/wails/pull/3445)
 - Fixed an issue where certain calls to createFrom in TypeScript would fail. Changed by [@twonull](https://github.com/twonull) in [PR](https://github.com/wailsapp/wails/pull/3435)
 - Fixed some typos in comments. Changed by [@reallylowest](https://github.com/reallylowest) in [PR](https://github.com/wailsapp/wails/pull/3357)
 - Fixed an issue where the destination file was not properly closed after copying. Changed by [@testwill](https://github.com/testwill) in [PR](https://github.com/wailsapp/wails/pull/3384)

